static/src/js/types/sharedTypes.ts

```diff
@@ -297,7 +297,16 @@ export type TimelineRequestParams = {
 }
 
 /** The request parameters for our request classes */
-export type RequestParams = TimelineRequestParams | CollectionRequestParams | ShapefileRequestParams
+/** The request parameters for a preferences request */
+export type PreferencesRequestParams = {
+  /** The preferences data */
+  preferences: Record<string, unknown>
```

Comment:
This should define whatever parameters we pass in the request, it isn't `unknown`